### PR TITLE
Allow construct_runtime to take cfg attributes for pallets

### LINKED ISSUE #10286: construct_runtime to allow for cfg macro for pallets
```rust
// Create the runtime by composing the FRAME pallets that were previously configured.
construct_runtime!(
	pub enum Runtime where
		Block = Block,
		NodeBlock = opaque::Block,
		UncheckedExtrinsic = UncheckedExtrinsic
	{
		#[cfg(feature = "develop")]
		AssetsRegistry : assets_registry::{Pallet, Call, Storage, Event<T>} = 55,
	}
);
```

why would i want it ?

- I want have single runtime, but able to make it different depending of feature gates.
- I do not want to copy paste list of other 50+ pallets in runtime

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #10286.

This PR allows `#[cfg]` attributes to exist on top of pallet declarations in the `construct_runtime!` macro, and it works exactly in the way that you would intuitively expect -- the pallet immediately following the attribute would exist in the runtime only if the feature gate as described by the `cfg` attribute is enabled, otherwise it does not exist.

### CODE REVIEW

**Comment by kianenigma** [comment]
wen `#cfg(test)` dispatchable? (https://github.com/paritytech/substrate/pull/11754)

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1745298

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1745299

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable-int 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1745310

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1745339

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1745340

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750310

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750312

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750387

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - cargo-check-benches 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750388

**Comment by shawntabrizi** [comment]
cc @sam0x17

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750445

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750446

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750447

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750510

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750509

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs. 
 The job name - test-linux-stable 
 The job logs - https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1750511

File: frame/support/test/tests/construct_runtime_ui/unsupported_pallet_attr.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic
+	{
+		System: system::{Pallet},
+		#[attr]
+		Balance: balances::{Config, Call},
+	}
+}
+
+fn main() {}
```
Comment (sam0x17) [inline frame/support/test/tests/construct_runtime_ui/unsupported_pallet_attr.rs:15]: missing newline

File: frame/support/test/tests/construct_runtime_ui/unsupported_meta_structure.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic
+	{
+		System: system::{Pallet},
+		#[cfg(feature(test))]
+		Balance: balances::{Config, Call},
+	}
+}
+
+fn main() {}
```
Comment (sam0x17) [inline frame/support/test/tests/construct_runtime_ui/unsupported_meta_structure.rs:15]: missing newline

File: frame/support/test/tests/construct_runtime_ui/invalid_meta_literal.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic
+	{
+		System: system::{Pallet},
+		#[cfg(feature = 1)]
+		Balance: balances::{Config, Call},
+	}
+}
+
+fn main() {}
```
Comment (sam0x17) [inline frame/support/test/tests/construct_runtime_ui/invalid_meta_literal.rs:15]: missing newline

**Review by sam0x17 — APPROVED** [review]
This makes sense to me and is a very reasonable improvement. I found some missing end-of-file newlines on a few files that would be cool to fix but other than that this looks good--macro code is well structured.

**Review by sam0x17 — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]
high level, non-expert macro review

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**